tests/test_client.py

```diff
@@ -0,0 +1,185 @@
+import pytest
+import testinfra
+import subprocess
+import os
+import time
+from datetime import datetime
+
+
+@pytest.fixture(scope='session')
+def client(request, foreman_version):
+    name = f"smoker-client-{request.param.replace('/', '_').replace(':', '_')}"
+
+    image_id = subprocess.check_output(['podman', 'images', '-q', '--filter', f"reference={request.param}"])
+    if not image_id:
+        subprocess.check_output(['sh', 'build_images.sh', foreman_version])
```

Comment:
This means that it can run `build_images.sh` for every client that's not found. Perhaps it should be structured as `build_image.sh` and pass an argument to avoid this?